test/EFCore.PG.FunctionalTests/MigrationsNpgsqlTest.cs

```diff
@@ -1626,6 +1626,82 @@ public virtual async Task Create_index_with_null_sort_order()
                 @"CREATE INDEX ""IX_People_FirstName_MiddleName_LastName"" ON ""People"" (""FirstName"" NULLS FIRST, ""MiddleName"", ""LastName"" NULLS LAST);");
         }
 
+        [Fact]
+        public virtual async Task Create_index_on_tsvector()
+        {
+            await Test(
+                builder => builder.Entity(
+                    "People", e =>
+                    {
+                        e.Property<int>("Id");
+                        e.Property<string>("FirstName").IsRequired();
+                        e.Property<string>("MiddleName").IsRequired();
+                        e.Property<string>("LastName").IsRequired();
+                    }),
+                builder => { },
+                builder => builder.Entity("People")
+                    .HasIndex("FirstName", "LastName")
+                    .UseToTsVector("simple")
+                ,
+                model =>
+                {
+                    var table = Assert.Single(model.Tables);
+                    var index = Assert.Single(table.Indexes);
+
+                    Assert.Equal("simple", index[NpgsqlAnnotationNames.IndexToTsVector]);
+
+                    Assert.Contains(index.Columns, i => i.Name == "FirstName");
+                    Assert.Contains(index.Columns, i => i.Name == "LastName");
+                });
+
+            AssertSql(
+                @"CREATE INDEX ""IX_People_FirstName_LastName"" ON ""People"" (to_tsvector('simple', ""FirstName"" || ' ' || ""LastName""));");
+        }
+
+        [Fact]
+        public virtual async Task Create_index_on_tsvector_with_nullable_columns()
```

Comment:
I think this can be combined with the above - just have a single model where one property is nullable and the other is non-nullable. See for example Add_column_generated_tsvector which I added, you can follow that.

Reply:
I doubted whether to divide these tests, or leave one. And at last I decided that they are still slightly different in meaning and divided them. But I don't mind to combine them and will do that.

Reply:
Done.